apps/emqx_rule_engine/src/emqx_rule_engine.erl

```diff
@@ -317,8 +318,14 @@ get_basic_usage_info() ->
         NumRules = length(EnabledRules),
         ReferencedBridges =
             lists:foldl(
-                fun(#{actions := Actions, from := From}, Acc) ->
-                    BridgeIDs0 = [BridgeID || <<"$bridges/", BridgeID/binary>> <- From],
+                fun(#{actions := Actions, from := Froms}, Acc) ->
+                    BridgeIDs0 =
+                        [
+                            BridgeID
+                         || From <- Froms,
+                            {ok, BridgeID} <-
+                                [emqx_bridge_resource:bridge_hookpoint_to_bridge_id(From)]
+                        ],
                     BridgeIDs1 = lists:filter(fun is_binary/1, Actions),
```

Comment:
This fold and filter can be extracted to a private func for better readability.

Reply:
I'll make this refactoring in a follow up PR.